File: 23-03-2024/Fibonacci series up to Nth term.java
//{ Driver Code Starts
import java.io.*;
import java.util.*;

class Main {
    // Driver code
    public static void main(String[] args) throws Exception {
        BufferedReader br =
            new BufferedReader(new InputStreamReader(System.in));
        int t = Integer.parseInt(br.readLine().trim());
        while (t-- > 0) {
            int n = Integer.parseInt(br.readLine().trim());
            Solution obj = new Solution();
            int ans[] = obj.Series(n);
            for (int i : ans) {
                System.out.print(i + " ");
            }
            System.out.println();
        }
    }
}
// } Driver Code Ends




class Solution {

    int[] Series(int n) {
      //Your Code Start
        int[] arr = new int[n+1];
        arr[0] = 0;
        arr[1] = 1;
        int i=2;
        int mod = (int)(1e9 + 7);
        while(i<n+1)
        {
            arr[i] = (arr[i-1] + arr[i-2]) % mod;
            i++;
        }
        return arr;
      //Your Code End

    }
}

File: 23-03-2024/Fibonacci series up to Nth term.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>
using namespace std;

// } Driver Code Ends
class Solution {
  public:
//Your Code Start
   const int mod = 1e9+7;
    vector<int> Series(int n) {
         vector<int>arr(n+ 1);
         
          if(n==1)
        {
            return {0,1};
        }
        
        arr[0]=0;
        arr[1]=1;
          
        
    for(int i=2;i<=n;++i)
    {
        arr[i]=(arr[i-1]+arr[i-2])%mod;
    }
    return arr;
      //Your Code End
        }
};


//{ Driver Code Starts.
int main() {
    int t;
    cin >> t;
    while (t--) {
        int n;
        cin >> n;
        Solution obj;

        vector<int> ans = obj.Series(n);
        for (auto x : ans) cout << x << " ";
        cout << "\n";
    }
    return 0;
}
// } Driver Code Ends
